zipkin-server/src/main/java/zipkin2/server/internal/brave/TracingConfiguration.java

```diff
@@ -103,7 +104,10 @@
       .serverSampler(new HttpSampler() {
         @Override public <Req> Boolean trySample(HttpAdapter<Req, ?> adapter, Req request) {
           String path = adapter.path(request);
-          return path.startsWith("/api") || path.startsWith("/zipkin/api");
+          if (path.startsWith("/api") || path.startsWith("/zipkin/api")) {
```

Comment:
Subtle, but the contract for the HttpSampler is to do your rate here (with a true or false) or return null to use the global rate. We were accidentally returning true which meant the global sampler was entirely ignored.

cc just as a lesson @anuraaga @kojilin @marcingrzejszczak